roles/validations/tasks/main.yml

```diff
@@ -56,5 +68,34 @@
       failed_when: not validation_exists.stat.exists
 
     - name: Run validations
-      ansible.builtin.include_tasks: "{{ item }}"
+      ansible.builtin.include_tasks:
+        file: xmlcreate.yml
       loop: "{{ cifmw_validations_list }}"
+
+- name: Get validations end time from system
+  ansible.builtin.command:
+    cmd: date +%Y-%m-%d-%H:%M:%S.%3N
+  register: validations_end_time
+
+- name: Calculate total validations run time
+  ansible.builtin.set_fact:
+    validations_run_time: "{{((validations_end_time.stdout | to_datetime('%Y-%m-%d-%H:%M:%S.%f')) - (validations_start_time.stdout | to_datetime('%Y-%m-%d-%H:%M:%S.%f'))).total_seconds()}}"
+
+- name: Write first line of xml results file
+  ansible.builtin.shell: |
+    mkdir -p "{{ cifmw_validations_xml_status_file_dir }}"
+    echo "<testsuite errors=\"{{ validations_errors }}\" failures=\"{{ validations_failed }}\" name=\"\" tests=\"{{ validations_executed }}\" time=\"{{ validations_run_time }}\">" >> {{ cifmw_validations_xml_status_file_dir }}/validations_results.xml
```

Comment:
Generating XML with bash is asking for trouble. There's not nice way to generate XML in Ansible, but using jinja2 would be a step forward.

Reply:
@pablintino Is this comment a goal or a requirement? Before I started this review I asked on the proj-create-integration-pipeline-jobs about creating xml files and the answer was:   it could be even a simple set of echo "" , but if you want to use something easy and more structured you could use https://pypi.org/project/junitparser/

Reply:
@drosenfe It's a suggestion to improve the maintainability of the role, specially given that I don't see a molecule test that ensure the rendered XML matches the expectations. If you and all the ones involved in this role are fine I'm ok with the change.

Reply:
I looked quite a lot at  junitparser. Doing a fake test from command line this is the output it gives:

<?xml version="1.0" encoding="utf-8"?>
<testsuites>
        <testsuite name="validations" tests="3" errors="0" failures="0" skipped="0" time="1.8">
                <testcase name="hugepages_and_reboot.yml" classname="validations.edpm" time="0.5"/>
                <testcase name="hotfix.yml" classname="validations.edpm" time="0.6"/>
                <testcase name="scaledown.yml" classname="validations.edpm" time="0.7"/>
        </testsuite>
</testsuites>

This different in a few ways from the current xml files generated by tempest

- it uses tabs instead of spaces (could be changed with sed after creation)
- it has an extra xml header (could be deleted after creation)
- the test cases have order: name, classname, time while tempest xml files have order: classname, name, time
- the testsuite order for tests, errors, failures, skipped is different

What does anyone think about using junitparser? There were some advantages. It sums the test cases, failures, and total time so that the validations roles does not need too track those.

Reply:
From an XML point of view:
* tabs vs spaces don't matter, they are all whitespaces (https://www.w3.org/TR/xml/#sec-common-syn)
* the XML header is correct, it doesn't need to be removed. It's part of valid XML files
* the order of attributes in an XML tag (so both testcase and testsuite) doesn't matter.
You don't even need to use pretty=True when exporting the XML file, the file is still a valid XML file (it can be rendered for humans by the viewer).

Reply:
Latest version uses junitparser to generate xml file